flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/AcrossCheckpointFileMergingSnapshotManager.java

```diff
@@ -31,12 +33,13 @@ public class AcrossCheckpointFileMergingSnapshotManager extends FileMergingSnaps
     private final PhysicalFilePool filePool;
 
     public AcrossCheckpointFileMergingSnapshotManager(
-            String id,
+            JobID jobId,
+            ResourceID tmResourceId,
```

Comment:
I'd suggest a id string for `FileMergingManager`. It only need a unique id without the knowledge of job or resource id.

Reply:
Thanks for this suggestion , I moved the generation of id to builder.